packages/rum/src/boot/recorderApi.ts

```diff
@@ -60,7 +60,10 @@ export function makeRecorderApi(startRecordingImpl: StartRecording): RecorderApi
       parentContexts: ParentContexts
     ) => {
       lifeCycle.subscribe(LifeCycleEventType.SESSION_RENEWED, () => {
-        if (state.status === RecorderStatus.IntentToStart) {
+        const status = state.status
+        stopStrategy()
+
+        if (status !== RecorderStatus.Stopped) {
           startStrategy()
         }
```

Comment:
```suggestion
        if (state.status === RecorderStatus.Started) {
          stopStrategy()
          state = { status: RecorderStatus.IntentToStart }
        }

        if (state.status === RecorderStatus.IntentToStart) {
          startStrategy()
        }
```
it seems to behave the same but I would find it easier to understand this way.
wdyt?